### PR TITLE
Transform API to unicode

### PR DESCRIPTION
**Pull request by ziima**

### CODE REVIEW

File: openid/kvform.py [hunk 1]
```diff
@@ -29,11 +36,15 @@ def err(msg):
...
     for k, v in seq:
-        if isinstance(k, types.StringType):
-            k = k.decode('UTF8')
-        elif not isinstance(k, types.UnicodeType):
-            err('Converting key to string: %r' % k)
-            k = str(k)
+        k = string_to_text(k, "Binary values for keys are deprecated. Use text input instead.")
+        v = string_to_text(v, "Binary values for values are deprecated. Use text input instead.")
+
+        if not isinstance(k, six.text_type):
```
Comment (tpazderka) [inline openid/kvform.py]: The order of operations should be reversed here. First, make sure to convert non string objects and then convert to text.

File: openid/oidutil.py [hunk 2]
```diff
@@ -148,3 +146,16 @@ def __hash__(self):
 
     def __repr__(self):
         return '<Symbol %s>' % (self.name,)
+
+
+def string_to_text(value, deprecate_msg):
+    """
+    Return input string coverted to text string.
+
+    If input is text, it is returned as is.
```
Comment (tpazderka) [inline openid/oidutil.py:159]: or any other type...

**Review by tpazderka — CHANGES_REQUESTED** [review]

File: openid/kvform.py (hunk 1 above)
Comment (ziima) [reply openid/kvform.py]: Fixed.

File: openid/oidutil.py (hunk 2 above)
Comment (ziima) [reply openid/oidutil.py:159]: That's not intended usage. I've added assert to warn about that.

**Review by tpazderka — APPROVED** [review]

**Pull request merged**